include/storage/shared_memory.hpp

```diff
@@ -30,6 +30,12 @@ namespace osrm
 namespace storage
 {
 
+enum class Ownership
```

Comment:
Yeah eventually all headers end up in each cpp file, but usually relying on indirect header includes is considered bad style.

There is actually nothing in `vector_view` that really is specific to shared memory. That whole area in the code base needs some proper naming refresh (we use the same code for process internal memory and shared memory segments...). There are only very few parts of the code that actually need to know about the shared memory stuff, but I guess this is pending a refactor. 😉